resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -2,58 +2,133 @@
     "Version": "2012-10-17",
     "Statement": [
         {
+            "Sid": "SnapShot",
             "Effect": "Allow",
             "Action": [
-                "ec2-instance-connect:SendSerialConsoleSSHPublicKey",
+                "ec2:DescribeSnapshots"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "CreateSnapShots",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:CreateSnapshots"
+            ],
+            "Resource": "arn:aws:ec2:*:*:instance/*"
+        },
+        {
+            "Sid": "DescribeSnapshotAttribute",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeSnapshotAttribute",
                 "ec2:CopySnapshot",
+                "ec2:CreateSnapshot"
+
+            ],
+            "Resource": "arn:aws:ec2:*:*:snapshot/*"
+        },
+        {
+            "Sid": "SendSerialConsoleSSHPublicKey",
+            "Effect": "Allow",
+            "Action": [
+                "ec2-instance-connect:SendSerialConsoleSSHPublicKey",
+                "ec2:GetSerialConsoleAccessStatus"
+            ],
+            "Resource": "*"
```

Comment:
Also, should these only be instances with red-hat-managed: true?

Reply:
Seems like it can't be restricted to just ec2 instances. resources "*" works

Reply:
An error occurred (AccessDeniedException) when calling the SendSerialConsoleSSHPublicKey operation: User: arn:aws:sts::722251204950:assumed-role/ManagedOpenShift-HCP-ROSA-Support-Role/5a9d529885b8d55fb4b1902fd87f9c2f is not authorized to perform: ec2-instance-connect:SendSerialConsoleSSHPublicKey on resource: arn:aws:ec2:us-east-1:722251204950:instance/i-fcaebklfreiuph98 because no identity-based policy allows the ec2-instance-connect:SendSerialConsoleSSHPublicKey action